### PR TITLE
git: Gui fixes

### PR DESCRIPTION
**Pull request by podsvirov**
Some fixes to allow using `git gui` command in `mingw` subsystems.

### CODE REVIEW

**Comment by podsvirov** [comment]
Hello @prati0100! Can your review this patch? Can we apply this changes to git-gui code base if it will work for others?

**Comment by dscho** [comment]
Thank you for working on this!

However, this is unfortunately the wrong repository: MSYS2-packages *never* take mingw-w64-* dependencies.

Also, the Git GUI patch should be made against the `master` branch of https://github.com/prati0100/git-gui and be submitted using [GitGitGadget](https://gitgitgadget.github.io/).

**Comment by podsvirov** [comment]
> Thank you for working on this!

It's working for you? Can you use `git gui` command from `mingw32/64` console afrer installing `git` and `mingw-w64-i686/x86_64-tk` packages?

> However, this is unfortunately the wrong repository: MSYS2-packages _never_ take mingw-w64-* dependencies.

It's no problem: we can drop optional dependencies from `PKGBUILD` file. But @Alexpux can we keep optional dependencies for user information.

> Also, the Git GUI patch should be made against the `master` branch of https://github.com/prati0100/git-gui and be submitted using [GitGitGadget](https://gitgitgadget.github.io/).

After testing in this distro, I will be create `PR` for appropriate repository.

**Comment by dscho** [comment]
> It's working for you?

I haven't tried, due to serious lack of time (and I still think that this is the wrong repository, MINGW-packages seems much more appropriate).

**Comment by podsvirov** [comment]
> > It's working for you?
> 
> I haven't tried, due to serious lack of time (and I still think that this is the wrong repository, MINGW-packages seems much more appropriate).

I understood (and it's not possible, because `git` available only in `MINGW-packages` :-) ).

**Comment by dscho** [comment]
That's right, because `git` is an MSYS2 package. Meaning: it links to the MSYS2 runtime, a POSIX emulation layer based on the Cygwin runtime.

But you claim to be interested in MINGW, i.e. a Git package that does _not_ rely on that POSIX emulation layer.

What you want, therefore, is the `mingw-w64-git` package, in https://github.com/msys2/MINGW-packages.

**Comment by podsvirov** [comment]
> That's right, because `git` is an MSYS2 package. Meaning: it links to the MSYS2 runtime, a POSIX emulation layer based on the Cygwin runtime.

All good.

> But you claim to be interested in MINGW, i.e. a Git package that does _not_ rely on that POSIX emulation layer.

No, because this `PR` about fixes to run `git gui` command. In this case `git` will run `git-gui` script, it's `tcl/tk` based application. To run `git-gui` script we need `wish` enterpreter, thats available in `mingw-w64-i686/x86_64-tk` package (in `mingw32/64` subsystem).

> What you want, therefore, is the `mingw-w64-git` package, in https://github.com/msys2/MINGW-packages.

No, I not want this now.

**Comment by lazka** [comment]
We can't really depend on mingw packages. We could handle the cases when they are missing gracefully and suggest to user to install them.

**Comment by podsvirov** [comment]
> We can't really depend on mingw packages. We could handle the cases when they are missing gracefully and suggest to user to install them.

Can you show examples of how to do this?

**Comment by podsvirov** [comment]
The mingw `tk` package required for working the `git gui` command from `mingw` shell. And optionaly `aspell` package from mingw or msys additionaly can be used by `git gui` to spellcheck the commit message.

**Comment by lazka** [comment]
Not sure. What happens if tk and aspell are missing and you call "git gui"?

**Comment by podsvirov** [comment]
> Not sure. What happens if tk and aspell are missing and you call "git gui"?

Something like:
```bash
$ git gui
/usr/lib/git-core/git-gui: line 10: exec: wish: not found
```

**Comment by podsvirov** [comment]
> We can't really depend on mingw packages.

The optional dependencies on mingw packages removed.

**Comment by podsvirov** [comment]
More questions, reviews, feedback?

If not, I propose to merge as is and end the discussion.

File: git/git-2.26.0-gui-fixes.patch [hunk 1]
```diff
@@ -0,0 +1,91 @@
+--- a/git-gui/git-gui.sh	2020-03-23 02:50:46.000000000 +0300
```
Comment (dscho) [inline git/git-2.26.0-gui-fixes.patch:1]: I am still of the rather strong opinion that this should be contributed to the Git GUI project by applying this patch on top of https://github.com/gitgitgadget/git's `git-gui/master`, then opening a PR there and submitting it to the Git mailing list for review via the `/submit` command in the PR.

The reason why I say this is that the MSYS2 maintainer (who could merge this patch here) can hardly be asked to review this patch touching idiosyncratic Tcl/Tk code.

And once it gets merged by @prati0100 (the Git GUI maintainer), it is just a matter of time for it to trickle down into v2.27.0 and then we do not need _any_ change here.

Comment (podsvirov) [reply git/git-2.26.0-gui-fixes.patch:1]: @dscho, done. Please, see https://github.com/gitgitgadget/git/pull/612. Can you help me to `/allow` on GitGitGadget?

Comment (dscho) [reply git/git-2.26.0-gui-fixes.patch:1]: Done.

**Comment by prati0100** [comment]
Fixed upstream: https://github.com/prati0100/git-gui/commit/88db24d724619d6c2e602c3b56016f59326d31e3

File: git/git-2.26.0-gui-fixes.patch [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
++
++######################################################################
++##
+ ## locate our library
+ 
+ if { [info exists ::env(GIT_GUI_LIB_DIR) ] } {
+ 	set oguilib $::env(GIT_GUI_LIB_DIR)
+ } else {
+-	set oguilib {@@GITGUI_LIBDIR@@}
++	if {[is_Cygwin]} {
```
Comment (lazka) [inline git/git-2.26.0-gui-fixes.patch:35]: imo this should be something like:

```tcl
proc is_MingwTcl {} {
	# Handle MSys2 which is only cygwin when MSYSTEM is MSYS.
	if {[info exists ::env(MSYSTEM)] && $::env(MSYSTEM) ne "MSYS"} {
		return 1;
	}
	return 0;
}

if {[is_Cygwin] && [is_MingwTcl]} {
...
```

to not break real cygwin

**Comment by lazka** [comment]
Commented on the upstream PR, and merged a simpler variant of this in #2886

so, closing.

**Comment by podsvirov** [comment]
@lazka , thanks for review, update and apply this suggestion!

**Pull request closed without merging**